<br>  

## <span style='color:blue'>Section 1: Import</span>  

In [1]:
import pandas as pd
import numpy as np
import copy

from sklearn.model_selection import train_test_split

<br>  

## <span style='color:blue'>Section 2: Read and drop index column</span>  

In [2]:
df_1 = pd.read_csv('../data/code_2_data.csv')

print('Before :', df_1.shape)
df_1.drop(columns=['index'], inplace=True)
print('After :', df_1.shape)

Before : (1177, 51)
After : (1177, 50)


<br>  

## <span style='color:blue'>Section 3: Drop row with missing value in outcome</span>  

In [3]:
print('Before :', df_1.shape)
df_1 = df_1.dropna(subset=['outcome'])
print('After :', df_1.shape)

Before : (1177, 50)
After : (1176, 50)


<br>  

## <span style='color:blue'>Section 4: Form X and y and check category distribution of y</span>  

In [4]:
X = df_1.drop(columns=['outcome'])
print('X :', X.shape)

y = df_1['outcome']
print('y :', y.shape)

X : (1176, 49)
y : (1176,)


In [5]:
y.value_counts(dropna=False, normalize=True)

0.0    0.864796
1.0    0.135204
Name: outcome, dtype: float64

In [6]:
y.value_counts(dropna=False)

0.0    1017
1.0     159
Name: outcome, dtype: int64

<br>  

## <span style='color:blue'>Section 5: Train test split</span>  

In [7]:
X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y, random_state=42, shuffle=True, test_size=0.25)

print('X train :', X_train.shape)
print('y train :', y_train.shape)
print('')

print('X test :', X_test.shape)
print('y test :', y_test.shape)

X train : (882, 49)
y train : (882,)

X test : (294, 49)
y test : (294,)


<br>  

## <span style='color:blue'>Section 6: Compare category distribution of y_train and y_test</span>  

In [8]:
y_train.value_counts(dropna=False, normalize=True)

0.0    0.865079
1.0    0.134921
Name: outcome, dtype: float64

In [9]:
y_train.value_counts(dropna=False)

0.0    763
1.0    119
Name: outcome, dtype: int64

In [10]:
y_test.value_counts(dropna=False, normalize=True)

0.0    0.863946
1.0    0.136054
Name: outcome, dtype: float64

In [11]:
y_test.value_counts(dropna=False)

0.0    254
1.0     40
Name: outcome, dtype: int64

<br>  

## <span style='color:blue'>Section 7: Merge X_train and y_train</span>  

In [12]:
df_train = copy.deepcopy(X_train)
df_train['outcome'] = y_train
df_train = pd.DataFrame(df_train)

<br>  

## <span style='color:blue'>Section 8: Merge X_test and y_test</span>  

In [13]:
df_test = copy.deepcopy(X_test)
df_test['outcome'] = y_test
df_test = pd.DataFrame(df_test)

<br>  

## <span style='color:blue'>Section 9: Save train and test datasets</span>  

In [14]:
df_train.to_csv('../data/code_3_train.csv', na_rep='NaN', index_label='index')
df_test.to_csv('../data/code_3_test.csv', na_rep='NaN', index_label='index')

<br>  
<br>  

## <span style='color:green'>Conclusions:</span>  

### <span style='color:green'>Testing dataset</span>  
* <span style='color:green'>Is now considered as unseen.</span>  

### <span style='color:green'>Section 6</span>  
* <span style='color:green'>Dataset imbalanced with dead ("1") at 13.5% and alive ("0") at 86.5%.</span>  
<br>  
<br>  


3 Jan 2022